pandas/tests/extension/decimal/test_decimal.py

```diff
@@ -205,6 +205,27 @@ def test_dataframe_constructor_with_dtype():
     tm.assert_frame_equal(result, expected)
 
 
+@pytest.mark.parametrize("frame", [True, False])
```

Comment:
why don't we add a test in base which raises NotImplementedError so that authors are forced to cover this?

Reply:
But is it useful to force EA authors to do it? This is basically checking the pandas dispatch (which we can do here), not the actual EA.astype implementation

Reply:
Or how about something like

```python
@pytest.mark.parametrize('typ, check', [
    ('category', 'is_categorical_dtype'),
    ...
])
def test_astype_category(self, data):
    assert check(self.astype(dtype))
```

Would that make sense as a base test? I think our default implementation would need to be updated to not fail that.

My main concern is that it would be difficult to override (e.g. skip) just some of the dtypes, so maybe we would have to write those as separate tests?

Reply:
I'm mildly concerned that EA authors will not correctly handle extension types. Our base implementation currently fails to handle them. Although we document it as `Cast to a NumPy array with 'dtype'.`. Should we make the return type Union[ndarray, ExtensionArray]`?

Reply:
I would leave such a test until we better figure out how to handle those cross EA/non-EA astype calls (the discussion we were having above in this PR )

> Should we make the return type Union[ndarray, ExtensionArray]?

Yeah, probably yes in any case (since EAs authors that provide multiple dtypes already do that, like IntegerArray)

Reply:
I would add an issue to make this test, not blocking this PR on it.